File: Task1.Service/Implementation/DepartmentService.cs
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Task1.Data.Models;
using Task1.Service.Interface;

namespace Task1.Service
{
    public class DepartmentService : IDepartmentService
    {
        private readonly Task1DbContext _context;
        public DepartmentService(Task1DbContext context)
        {
            _context = context;
        }

        public async Task<IEnumerable<Department>> GetDepartments()
        {
            return await _context.Departments.ToListAsync();
        }

        public async Task<Department> GetDepartmentById(int id)
        {
            return await _context.Departments.FindAsync(id);
        }
        public async Task CreateDepartment(Department department)
        {
            _context.Departments.Add(department);
            await _context.SaveChangesAsync();
        }

        public async Task UpdateDepartment(Department department)
        {
            _context.Update(department);
            await _context.SaveChangesAsync();
        }

        public async Task DeleteDepartment(int id)
        {
            var department = await _context.Departments.FindAsync(id);
            if (department != null)
            {
                _context.Departments.Remove(department);
                await _context.SaveChangesAsync();
            }
        }

        
    }

    
    
}

File: Task1/wwwroot/ScriptNG/Services/DepartmentService.js
ngApp.service("departmentService", function ($http) {

    this.GetDepartments = function () {
        return $http.get('/Departments/GetDepartments');
    };

    this.CreateDepartment = function (data) {
        return $http.post('/Departments/Create', data);
    };

    this.EditDepartment = function (id, data) {
        return $http.post('/Departments/Edit/' + id, data);
    };

    this.DeleteDepartment = function (id) {
        return $http.delete('/Departments/Delete/' + id);
    };

});

